### ISSUE TITLE
Update @lingui/vite-plugin to support Vite 5

### ISSUE THREAD

**Issue by ptmkenny** [lingui/js-lingui#1805]
**Describe the bug**
[Vite 5 was released on November 16, 2023](https://vitejs.dev/blog/announcing-vite5).

`@lingui/vite-plugin` can't currently be used with Vite 5.

**To Reproduce**

Try to update to vite 5 `npm update`:

```
npm ERR! Found: vite@5.0.2
npm ERR! node_modules/vite
npm ERR!   dev vite@"^5" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer vite@"3 - 4" from @lingui/vite-plugin@4.5.0
npm ERR! node_modules/@lingui/vite-plugin
npm ERR!   dev @lingui/vite-plugin@"^4" from the root project
npm ERR! 
```

**Expected behavior**
Lingui can be used with Vite 5.

**Comment by thekip** [comment]
Did you test it ignoring the warning? I'm wondering do i need simply expand the constraint or some other actions needed?

**Comment by ptmkenny** [comment]
Ok, I got my app building, and it seems to be that the constraint just needs to be incremented.

I can switch between languages in m app and everything appears fine.

**Comment by evertbouw** [comment]
With an upgrade to Vite 5.0 I'm getting errors on dev server startup `No loader is configured for ".po" files`
prod builds working fine

**Comment by evertbouw** [comment]
Turns out this is from a change in esbuild https://github.com/evanw/esbuild/releases/tag/v0.19.0
So I guess this is the fix https://github.com/evertbouw/lingui-issue/commit/d23a670869d5b2d8b33b395a2392fdfbc0b24914

**Comment by evertbouw** [comment]
Esbuild changed something for dynamic imports https://github.com/evanw/esbuild/releases/tag/v0.19.0
So I guess this is the fix now https://github.com/evertbouw/lingui-issue/commit/8aed6a0b8bd7584ef6436aae66fb1f789975ffaa

**Comment by thekip** [comment]
yeah it seems this is the way. I also remember that i already did it in one of the examples, but quick check in the repo didn't show that.

**Comment by renchap** [comment]
An alternative is to use `import('.../file.po?lingui')`, then it works properly (in my app at least)

**Comment by akiver** [comment]
Hi, I also had this issue with npm 9 which is strict about peer deps:
<img width="650" alt="Screenshot 2023-12-13 at 23 31 02" src="https://github.com/lingui/js-lingui/assets/1842524/6e5e5eea-92be-4288-9486-19b125020d9f">

As a tmp workaround I added a [.npmrc](https://github.com/akiver/cs-demo-manager/blob/main/.npmrc) file with `legacy-peer-deps = true` to force npm to ignore peer deps issues.

About the loader error, I used the @renchap fix when [importing](https://github.com/akiver/cs-demo-manager/blob/main/src/ui/bootstrap/locale-provider.tsx#L19) po files - it works fine.

**Comment by monholm** [comment]
The `No loader is configured for ".po" files` error was fixed in `vite@5.0.9`. I've just opened a PR that widens the peer dependency version range, as we haven't experienced any other issues using `vite@5` :)
